src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4922,6 +4922,11 @@ def create_functionapp(cmd, resource_group_name, name, storage_account, plan=Non
         functionapp_def.kind = 'functionapp'
         # if os_type is None, the os type is windows
         is_linux = bool(os_type and os_type.lower() == LINUX_OS_NAME)
+        plan_name = generatePlanName(resource_group_name)
```

Comment:
I think this is the reason why the recording tests failed. A new random plan name will be generated when each time this function is run. Thus, when tests are run under recording mode, the plan name is generated again and different from what in the recoding file.
https://github.com/Azure/azure-cli/blob/900d76206b748f84864a9a4f441dcdd53ea175ff/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L5394-L5398

Reply:
Hi @yanzhudd We actually do intend to create a new plan resource with a new name on behalf of the customer.  Does this mean we should just make the tests run live? Or what would you require for this PR to be merged?

Reply:
yes, please mark the related tests as live only, and remove the corresponding recording files as well.